watertap/unit_models/zero_order/gas_sparged_membrane_zo.py

```diff
@@ -0,0 +1,308 @@
+        self._perf_var_dict["Mass flow of gas extracted (kg/s))"] = self.flow_mass_gas_extraction
+
+        # Add performance constraints
+        # Water recovery
+        @self.Constraint(self.flowsheet().time, doc='Water recovery equation')
+        def water_recovery_equation(b, t):
+            return (b.recovery_frac_mass_H2O[t] *
+                    b.properties_in[t].flow_mass_comp["H2O"] ==
+                    b.properties_treated[t].flow_mass_comp["H2O"])
+
+        # Flow balance
+        @self.Constraint(self.flowsheet().time, doc='Overall flow balance')
+        def mass_balance(b, t):
+            return (sum(b.properties_in[t].flow_mass_comp[j] for j in self.config.property_package.component_list) ==
+                    sum(b.properties_treated[t].flow_mass_comp[j] for j in self.config.property_package.component_list) +
+                    sum(b.properties_byproduct[t].flow_mass_comp[j] for j in self.config.property_package.component_list) +
+                    b.flow_mass_gas_extraction[t])
+
+        # Gas extraction
+        @self.Constraint(self.flowsheet().time,
+                         doc='Gas extraction equation')
+        def mass_gas_extraction_equation(b, t):
+            return (b.flow_mass_gas_extraction[t] == b.gas_mass_influent_ratio[t]
+                    * sum(b.properties_in[t].flow_mass_comp[j]
+                          for j in self.config.property_package.component_list))
+
+        # Solute removal
+        @self.Constraint(self.flowsheet().time,
+                         self.config.property_package.solute_set,
+                         doc='Solute removal equations')
+        def solute_removal_equation(b, t, j):
+            return (b.removal_frac_mass_solute[t, j] *
+                    b.properties_in[t].flow_mass_comp[j] ==
+                    b.properties_byproduct[t].flow_mass_comp[j])
+
+        # Solute concentration of treated stream
+        @self.Constraint(self.flowsheet().time,
+                         self.config.property_package.solute_set,
+                         doc='Constraint for solute concentration in treated '
+                         'stream.')
+        def solute_treated_equation(b, t, j):
+            return ((1 - b.removal_frac_mass_solute[t, j]) *
+                    b.properties_in[t].flow_mass_comp[j] ==
+                    b.properties_treated[t].flow_mass_comp[j])
+
+        self._stream_table_dict = {"Inlet": self.inlet,
+                                   "Treated": self.treated,
+                                   "Byproduct": self.byproduct}
+
+        self._perf_var_dict["Water Recovery"] = self.recovery_frac_mass_H2O
+        self._perf_var_dict["Solute Removal"] = self.removal_frac_mass_solute
+
+        self._get_Q = _get_Q_gas_extraction
+
+        self._Q = Reference(self.properties_in[:].flow_vol)
+        pump_electricity(self, self._Q)
+
+def initialize_gas_extraction(blk, state_args=None, outlvl=idaeslog.NOTSET,
```

Comment:
Does this differ from the one in the SIDO method, and same for the scaling method below? If not, you could just import the method from there instead to save duplicated code (and automatically get any updates made to those).

Reply:
There's one addition to the scaling method, but I can do that for the initialization method.